lib/base/networking/api/grpc_handler.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:gocast_mobile/base/networking/api/gocast/api_v2.pbgrpc.dart';
+import 'package:gocast_mobile/models/error_model.dart';
+import 'package:gocast_mobile/models/token_model.dart';
+import 'package:grpc/grpc.dart';
+
+/// Handles gRPC communication for the application.
+class GrpcHandler {
+  final String host;
+  final int port;
+  late ClientChannel _channel;
+
+  /// Creates a new instance of the `GrpcHandler` class.
+  ///
+  /// The [host] and [port] are required.
+  GrpcHandler(this.host, this.port) {
+    debugPrint('Connecting to gRPC server at $host:$port');
+
+    _channel = ClientChannel(
+      host,
+      port: port,
+      options: const ChannelOptions(credentials: ChannelCredentials.insecure()),
+    );
+  }
+
+  /// Shuts down the gRPC client channel.
+  Future<void> shutdown() async {
+    await _channel.shutdown();
+  }
+
+  /// Calls a gRPC method.
+  ///
+  /// This method takes a [grpcMethod] function that makes the actual gRPC call.
+  /// It handles loading the JWT token, setting up the call options, and error handling.
+  ///
+  /// Throws an [AppError] if a network error occurs or if the gRPC call fails.
+  Future<T> callGrpcMethod<T>(
+    Future<T> Function(APIClient client) grpcMethod,
+  ) async {
+    try {
+      String token = await Token.loadToken('jwt');
```

Comment:
Maybe for robustness we check the token was successfully reveived ?

```Dart
if (token == null || token.isEmpty) {
  throw AppError.authenticationError('JWT token is missing');
}
```

Reply:
The token is always not null as the `Token.loadToken` method already throws an `AppError.authenticationError()` if no JWT token is set.
![image](https://github.com/TUM-Dev/gocast_mobile/assets/106114526/3cec8e37-9c86-474e-9dac-deae52fbe7e5)

(Checking if the token is an empty string could be implemented in `Token.loadToken` - however, this would not be an issue as an empty token would be rejected by the API anyways)